lib/compress.js

```diff
@@ -4655,4 +4655,36 @@ merge(Compressor.prototype, {
     OPT(AST_PrefixedTemplateString, function(self, compressor){
         return self;
     });
+
+    OPT(AST_ObjectKeyVal, function(self, compressor){
+        // p:function(){} ---> p(){}
+        // p:function*(){} ---> *p(){}
+        // p:async function(){} ---> async p(){}
+        // p:()=>{} ---> p(){}
```

Comment:
Basically take this:
https://github.com/mishoo/UglifyJS2/blob/5ea1da2d42d8063028b8178df1174f9752ab2c22/lib/compress.js#L3444-L3458

... and remove this part:
https://github.com/mishoo/UglifyJS2/blob/5ea1da2d42d8063028b8178df1174f9752ab2c22/lib/compress.js#L3446-L3452

... then you should have `this` detection.